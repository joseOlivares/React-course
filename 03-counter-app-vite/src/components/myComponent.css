


.customText{
    color: #61dafb;
}