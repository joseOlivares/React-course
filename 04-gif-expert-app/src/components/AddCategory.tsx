import { useState } from "react"


interface AddCategoryProps { //creamos un ainterfaz que le diga a Typescript que onNewCategory es una function
    onNewCategory: (category: string) => void;
    onClearCategories: () => void;
}

export const AddCategory = ({ onNewCategory, onClearCategories }: AddCategoryProps) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        const newValue = inputValue.trim();
        if(newValue.length <= 1) return; //si estÃ¡ vacio no se agrega

        console.log(newValue);
        onNewCategory(newValue);
        //setCategories((categories: string[]) => [...categories, inputValue]);
        setInputValue(''); //limpiamos el input
    }

 

  return (
    <form onSubmit={(event) => onSubmit(event)}>
        <div className="row mt-3">
            <div className="col">
                <input type="text" className="form-control"  placeholder="Buscar gifs" value={inputValue} onChange={onInputChange}/>
            </div>
            <div className="col-auto">
                <button className="btn btn-primary"  onClick={onClearCategories}>Limpiar</button>
            </div>

        </div>       
    </form>


    )
}
