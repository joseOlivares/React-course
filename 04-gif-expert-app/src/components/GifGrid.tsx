
import { tenorApiKey } from "../apikey/tenorApiKey";

interface GifGridProps {
  category: string;
}

export const GifGrid = ({ category }: GifGridProps) => {


const getGifs = async() => {
  try {
    const url = `https://tenor.googleapis.com/v2/search?q=${category}&key=${tenorApiKey}&limit=10`;
    const resp = await fetch(url);
    const data = await resp.json();
    console.log(data); // log the response data to the console
    const gifs = data.results.map((img: any) => {
      return {
        id: img.id,
        title: img.title,
        url: img.media_formats.gif.url,
        description: img.content_description
      }
    })
    console.log(gifs);
  } catch (error) {
    console.error(error); // log any errors to the console
  }
}

    getGifs();

  return (
    <div className="row">
        <h3>{category}</h3>
    </div>
  )
}
