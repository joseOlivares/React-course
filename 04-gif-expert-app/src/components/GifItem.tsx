
interface GifItemProps {
    id: string;
    title: string;
    url: string;
    description?: string;
    tag: string;
}
  


export const GifItem = (image: GifItemProps) => {
  return (
    <div className="card">
      <img src={image.url} alt={image.tag} className="card-image object-fit-scale" />
      <p className="card-text text-center">{image.tag}</p>
    </div>

  )
}
