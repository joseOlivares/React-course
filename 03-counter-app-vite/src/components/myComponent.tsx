import './myComponent.css'



//Fragment nos ayuda a resolver el problema del contenedor Padre que es obligatorio

//<> </> le dice a React que es un Fragment
    const newMesg="Variable JavaSCript";
    const arrayObj=[{name:"pepe",age:20},{name:"Maria",age:16}];

    const getSuma=(a:number,b:number)=>{
        return a+b;
    }

    interface Props {
        name: string;
        title: string;
        secretNum?: number;
    };


export const MyComponent = ({name, title, secretNum}: Props) => {




    return (
        <> 
            <h1 className="customText">{title}</h1>
            <h2>Name: {name}</h2>
            <h2>Secret Num:{secretNum}</h2>
            <p>Esto es un parrafo</p>
            <h2>{getSuma(7,5)}</h2>
            <h3>{newMesg}</h3>

    {arrayObj.map((element, index) => (
      <div key={index}>{element.name} - {element.age}</div>
    ))}
        </>
    );
};
