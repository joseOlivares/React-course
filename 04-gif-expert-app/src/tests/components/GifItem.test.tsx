import { render, screen } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('GifItem Component', () => {
  const props = {
    id: '123',
    title: 'Test Title',
    url: 'https://test-url.com/image.jpg',
    tag: 'Test Tag'
  };

  test('should render correctly', () => {
    render(<GifItem {...props} />);
    
    // Verificar que la imagen se renderiza con el URL correcto
    const image = screen.getByRole('img');
    expect(image).toHaveAttribute('src', props.url);
    expect(image).toHaveAttribute('alt', props.tag);
    
    // Verificar que el texto del tag se muestra
    expect(screen.getByText(props.tag)).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const { container } = render(<GifItem {...props} />);
    expect(container).toMatchSnapshot();
  });
});