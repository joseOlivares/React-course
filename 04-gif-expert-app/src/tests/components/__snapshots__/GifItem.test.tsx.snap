// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`GifItem Component should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Tag"
      class="card-image object-fit-scale"
      src="https://test-url.com/image.jpg"
    />
    <p
      class="card-text text-center"
    >
      Test Tag
    </p>
  </div>
</div>
`;
